### PR TITLE
feat: queriable org datasource

### PR DESCRIPTION
**Pull request by eliobischof**
Tests are here: #111

### CODE REVIEW

**Review by stebenz — COMMENTED** [review]
Looks good, we could also have a look at splitting the datasource into two, one for getByID and one for filtering with different queries where you could also receive multiple.
Example: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnet and https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets

**Comment by eliobischof** [comment]
> Looks good, we could also have a look at splitting the datasource into two, one for getByID and one for filtering with different queries where you could also receive multiple. Example: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnet and https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/subnets

I like the suggestion an split the data source int zitadel_org and zitadel_orgs. Can you please review again. I also adjusted the tests that are now also reviewable #111. Thanks @stebenz

File: zitadel/v2/org/funcs.go [hunk 1]
```diff
@@ -82,47 +74,92 @@ func update(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
...
+	resp, err := client.ListOrgs(ctx, req)
+	if err != nil {
+		return diag.Errorf("error while getting org by id %s: %v", orgName, err)
+	}
+	orgIDs := make([]string, len(resp.Result))
+	for i, org := range resp.Result {
+		orgIDs[i] = org.Id
+	}
+	// If the ID is blank, the datasource is deleted and not usable.
+	d.SetId("-")
```
Comment (stebenz) [inline zitadel/v2/org/funcs.go:163]: Did you test that if the list of orgs changes that the logic with for each gets managed correctly with terraform?

Comment (eliobischof) [reply zitadel/v2/org/funcs.go:163]: I tested this config:

```hcl
data zitadel_orgs list {
  domain = "policy"
  domain_method = "TEXT_QUERY_METHOD_CONTAINS_IGNORE_CASE"
}

data zitadel_org get {
  for_each = toset(data.zitadel_orgs.list.ids)
  id = each.value
}

output "name" {
  value = toset([
    for org in data.zitadel_org.get : org.name
  ])
}
```

When adding an org matching the criteria, the new org is added to the plan:

```
tf apply
data.zitadel_orgs.list: Reading...
data.zitadel_orgs.list: Read complete after 0s [id=-]
data.zitadel_org.get["227660920908677123"]: Reading...
data.zitadel_org.get["227657510687866883"]: Reading...
data.zitadel_org.get["227777495313678339"]: Reading...
data.zitadel_org.get["227657510687866883"]: Read complete after 0s [id=227657510687866883]
data.zitadel_org.get["227660920908677123"]: Read complete after 0s [id=227660920908677123]
data.zitadel_org.get["227777495313678339"]: Read complete after 0s [id=227777495313678339]

Changes to Outputs:
  ~ name = [
        # (1 unchanged element hidden)
        "domain-policy-org-jfftgbbewx",
      + "new-policy",
    ]

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

name = toset([
  "domain-policy-org-dxmnvkusoo",
  "domain-policy-org-jfftgbbewx",
  "new-policy",
])
```

**Review by stebenz — APPROVED** [review]

**Pull request merged**